### ISSUE TITLE
disable() for textInput doesn't work in Rmd 

### ISSUE THREAD

**Issue by richarddmorey** [daattali/shinyjs#147]
I'm trying to use disable() to disable a shiny textInput in an interactive Rmd document. Here's a MWE: 

[MWE on GitHub gist](https://gist.githubusercontent.com/richarddmorey/6e1e3123e53faf1a620d63ea68a1844b/raw/4c8ffb04bf98257527f40ae56e6e648ca54e52c4/index.Rmd)

I expected the "ID" textInput to be disabled on load and contain the text "Updated.", but it does not disable. It does, however, contain the text "Updated.", so I know that code block is running.

Here's my session info:

```
> sessionInfo()
R version 3.4.0 (2017-04-21)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] shinyjs_0.9.1

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.13      digest_0.6.12     rprojroot_1.2     mime_0.5         
 [5] R6_2.2.2          xtable_1.8-2      jsonlite_1.5      backports_1.1.0  
 [9] magrittr_1.5      evaluate_0.10.1   stringi_1.1.5     miniUI_0.1.1     
[13] rmarkdown_1.6     flexdashboard_0.5 tools_3.4.0       stringr_1.2.0    
[17] shiny_1.0.5       httpuv_1.3.5      yaml_2.1.14       rsconnect_0.8    
[21] compiler_3.4.0    htmltools_0.3.6   knitr_1.17       
```

**Comment by daattali** [comment]
The code block is indeed running, the problem is that it runs before the text input even exists. I would say this is the correct behaviour, because shinyjs should not keep a memory of "if this element doesn't exist right now, I should disable it in the future when it exists", that would cause a lot more unexpected behaviours (I have thought about this and this is the intended behaviour).

It looks like you're trying to disable the the input on page load, so you can wrap the input in a `disabled()` function to achieve what you want, it will be disabled immediately and can be enabled later.

The observer code will work if it runs after the input is created. For example if you add an action button and make the observer run when the button is pressed, then the input will get disabled.

**Comment by richarddmorey** [comment]
What I'm eventually trying to achieve is disabling it using a variable in the query string. Say, if "active=0" exists in the query string, then disable the text box. The problem is that the `textInput()` call is outside a reactive context, but the query string has to be read inside a reactive context.

Something like this:
[MWE on GitHub gist](https://gist.githubusercontent.com/richarddmorey/6e1e3123e53faf1a620d63ea68a1844b/raw/9e6d0a970667d659fe3b42cd9a5e6fa823f6640e/index2.Rmd)

shows what I want to achieve: read the query string, fill in the default value, then deactivate the input if asked. Do you see a way around this?

**Comment by daattali** [comment]
Do you mind posting on stackoverflow since this is more of a question than
a bug? And that way others can help :) I have a few projects due next week
and can't help with this right now, sorry!

On Nov 29, 2017 4:29 PM, "Richard Morey" <notifications@github.com> wrote:

What I'm eventually trying to achieve is disabling it using a variable in
the query string. Say, if "active=0" exists in the query string, then
disable the text box. The problem is that the textInput() call is outside a
reactive context, but the query string has to be read inside a reactive
context.

Something like this:
MWE on GitHub gist
<https://gist.githubusercontent.com/richarddmorey/6e1e3123e53faf1a620d63ea68a1844b/raw/9e6d0a970667d659fe3b42cd9a5e6fa823f6640e/index2.Rmd>

shows what I want to achieve: read the query string, fill in the default
value, then deactivate the input if asked. Do you see a way around this?

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/daattali/shinyjs/issues/147#issuecomment-348002244>,
or mute
the thread

**Comment by richarddmorey** [comment]
will do.

**Comment by daattali** [comment]
Thank you

On Nov 30, 2017 05:51, "Richard Morey" <notifications@github.com> wrote:

Closed #147 <https://github.com/daattali/shinyjs/issues/147>.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/daattali/shinyjs/issues/147#event-1365374991>, or mute
the thread
